### PR TITLE
[Docs] [Train] Update Train API reference and docs

### PR DESCRIPTION
**Pull request by amogkam**
Signed-off-by: Amog Kamsetty <amogkamsetty@yahoo.com>

Adds back more Ray Train APIs to Ray Train docs.

Also makes updates to the user guide for better references.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/api.rst [hunk 1]
```diff
@@ -7,6 +7,73 @@ This page covers advanced configurations for specific frameworks using Train.
 
 For different high level trainers and their usage, take a look at the :ref:`AIR Trainer package reference <air-trainer-ref>`.
 
+.. _train-api-torch-utils:
```
Comment (richardliaw) [inline doc/source/train/api.rst:10]: I thought these get captured in https://github.com/ray-project/ray/pull/28192/files#diff-c096028dc439275ded6e03c349bd56006c31319501b9bbc46d68994006aa0db2R255-R258 ?

Comment (amogkam) [reply doc/source/train/api.rst:10]: Idts- these are not part of the AIR session API

Comment (richardliaw) [reply doc/source/train/api.rst:10]: then why do we need :no-index:?

Comment (amogkam) [reply doc/source/train/api.rst]: Updated the PR, PTAL!

**Review by Yard1 — APPROVED** [review]
Thanks, love it!

**Review by richardliaw — APPROVED** [review]

**Comment by amogkam** [comment]
Failing tests are fixed on master

**Pull request merged**